src/service.ts

```diff
@@ -92,10 +97,16 @@ export class TaskService {
           .catch((error) => this.logger.warn(`Stopping activity failed`, { activityId: activity.id, error })),
       ),
     );
-    this.logger.info("Task Service has been stopped");
+    this.logger.info("Task Service has been stopped", {
+      stats: {
+        retryCount: this.retryCount,
+      },
+    });
   }
 
   private async startTask(task: Task) {
+    task.onStateChange((state) => this.logger.debug("Task changed state", { taskId: task.id, state: state }));
```

Comment:
it's better to put it in the `run()` method, there is already this handler there